### ISSUE TITLE
After 9.4.X upgrade Unified Alerting "Query & Results" shows error

### ISSUE THREAD

**Issue by tchameroy** [grafana/grafana#64354]
**What happened**:
After upgrading from 9.3.0 to 9.4.1(Now on 9.4.3 and issue still persists) the Query & Results drop down on the Alert View page shows the following error for Alert Rules using InfluxDB(InfluxQL) and Prometheus Datasources(These are the only two datasources we are using and can test with).

```
Query not available
Cannot display the query preview. Some of the data sources used in the queries are not available.
```
<img width="800" alt="Screenshot 2023-03-07 at 12 16 31 PM" src="https://user-images.githubusercontent.com/28870027/223527813-983b0385-8f10-4bc5-b988-416a52fe4ebc.png">

The Alert Rule's query has the correct datasource type along with the correct datasource uid(omitted) which is working normally
```
datasource:
  type: influxdb
  uid: omitted
dsType: influxdb
```

**What you expected to happen**:
A preview of the query should display

**How to reproduce it (as minimally and precisely as possible)**:
Navigate to Alert Rules -> Select an alert -> Click the "view" button -> Click Query & Results drop down -> Preview should show but it does not

**Anything else we need to know?**:
N/A

**Environment**:
- Grafana version: 9.4.3
- Data source type & version: InfluxDB(InfluxQL) & Prometheus
- OS Grafana is installed on: CentOS
- User OS & Browser: N/A
- Grafana plugins: N/A
- Others:

**Comment by jostrasser** [comment]
Hi,
I am experiencing the same issue. 
Only reproducible on migrated / older rules. 
If I edit the rule and run a "Preview" the data is shown correctly.
If I create a fresh rule all is working as expected.

Datasource is also influxdb (1.8.x)
Grafana version 9.3.8

**Comment by tonypowa** [comment]
:wave: Thanks for creating this issue, @jostrasser 

i am adding this issue to the alerting squad's board for review

**Comment by malmor** [comment]
Same issue here - running Grafana v9.3.8 (upgraded from v9.3.6).
In our case the datasource is not InfluxDB but Prometheus (v2.42.0)

**Comment by maxwellvarner** [comment]
I too am experiencing this issue. It was working on my migrated panels from grafana v7.5.1 to v9.3.4. However, after I upgraded to v9.4.3 yesterday these previews stopped working on those migrated panels. 

This same error message displays for alert rules that were created in v9.3.4 after the upgrade to v9.4.3.

As mentioned above, newly created alerts display this preview without issue in v9.4.3 in the Query & Results dropdown.

**Comment by mmusenbr** [comment]
We have the same problem, but we use provisioning to create our alerts. I tried to re-create on alert manually, export it and compare it to our provisioning. There were some slight changes but it seems to be this one: previously the datasource for the condition (in our case classical_condition) was defined with: `datasourceUid: "-100"`, on the newly created exported one I get `datasourceUid: __expr__`. I will try to re-provisioning with only that change and come back.

**Comment by mmusenbr** [comment]
> ...  I will try to re-provisioning with only that change and come back.
Did a simple search and replace for the `datasourceUid: "-100"` with `datasourceUid: __expr__`. Graphs are back for us.

Would also explain 'how' the workaround proposed by @maxwellvarner works.

**Comment by mmusenbr** [comment]
Seems to be related to this change https://github.com/grafana/grafana/pull/62510, which got backported to 9.0, 9.1, 9.2, 9.3. Would explain also non 9.4 reports.

**Comment by itsmylife** [comment]
Thanks for reporting this. This has been fixed in `v9.4.x` (See: https://github.com/grafana/grafana/pull/64235) but it hasn't been released. And we realized that it wasn't backported to `v9.3.x`. We just created a backport. See: https://github.com/grafana/grafana/pull/64949

This will fix the issues.

**Comment by soniaAguilarPeiron** [comment]
Closing after merging the [backport to v.9.3.x](https://github.com/grafana/grafana/pull/64949).
